scripts/build/check_init_priorities.py

```diff
@@ -75,9 +76,18 @@ class Priority:
     """
     def __init__(self, name):
         for idx, level in enumerate(_DEVICE_INIT_LEVELS):
+            expr = re.compile(rf".*{level}(\d+)_(\d+).*")
             if level in name:
                 _, priority_str = name.strip("_").split(level)
-                priority, sub_priority = priority_str.split("_")
+
+                # ARCMWDT linker generates extended section names
+                # (ex.: .rela.z_init_POST_KERNEL40_0_.__init_k_sys_work_q_init).
+                # limitation is required for reading only 2 parameters from
+                # priority string.
```

Comment:
this comment is now redundant, also please check @fabiobaltieri PR, because these changes may not be needed at all

Reply:
@fabiobaltieri,  @gmarull ,  Suggested changes https://github.com/zephyrproject-rtos/zephyr/pull/62459 works fina for WMDT compiler. So if pointed PR is applied in nearest future there is no need in this one (https://github.com/zephyrproject-rtos/zephyr/pull/62459).

Reply:
That's merged, hopefully we should be good to go

Sorry about the breakage again, thanks for you patience.